onnxruntime/core/providers/qnn/builder/qnn_model.h

```diff
@@ -55,22 +52,31 @@ class QnnModel {
   Status ParseGraphInputOrOutput(ConstPointerContainer<std::vector<NodeArg*>>& input_output_defs,
                                  std::unordered_map<std::string, OnnxTensorInfo>& input_output_info_table,
                                  std::unordered_map<std::string, size_t>& input_output_index,
-                                 std::unordered_map<std::string, size_t>& input_output_index_without_initializers,
                                  bool is_input = false);
 
   const std::unordered_set<std::string>& GetInitializerInputs() const { return initializer_inputs_; }
   bool IsGraphInitializerInput(const std::string input_name) {
     return initializer_inputs_.find(input_name) != initializer_inputs_.end();
   }
 
-  size_t GetInputIndex(const std::string& name) const {
+  // Return the input index within Ort graph which has initializers included
+  size_t GetOrtInputIndex(const std::string& name) const {
     return GetInputOutputIndex(name, inputs_info_);
   }
 
+  // Return the pure input index which doesn't cover initializers
+  size_t GetGraphInputIndex(const std::string& name) const {
+    auto it = model_input_index_map_.find(name);
+    ORT_ENFORCE(it != model_input_index_map_.end(), "Input name not found.");
+    return it->second;
+  }
+
   size_t GetOutputIndex(const std::string& name) const {
     return GetInputOutputIndex(name, outputs_info_);
   }
 
+  Status DeserializeGraphInforFromBinaryInfo(const QnnSystemContext_GraphInfo_t& qnn_sys_ctx_graph_info);
```

Comment:
```suggestion
  Status DeserializeGraphInfoFromBinaryInfo(const QnnSystemContext_GraphInfo_t& qnn_sys_ctx_graph_info);
```

Not sure if the `Infor` is intentional.